pandas/tests/indexes/multi/test_reshape.py

```diff
@@ -150,6 +150,16 @@ def test_append_index():
     tm.assert_index_equal(result, expected)
 
 
+@pytest.mark.parametrize("name, exp", [("b", "b"), ("c", None)])
+def test_append_names_match(name, exp):
+    # GH#48288
+    midx = MultiIndex.from_arrays([[1, 2], [3, 4]], names=["a", "b"])
+    midx2 = MultiIndex.from_arrays([[3], [5]], names=["a", name])
```

Comment:
So if names completely mismatch here e.g. `names=[name, "a"]` then the result names should be `[None, None]`? And do we have a test for this?

Reply:
Yes, and no looks like I forgot this